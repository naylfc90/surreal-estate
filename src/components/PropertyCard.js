import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBed, faShower, faEnvelope } from "@fortawesome/free-solid-svg-icons";

import "../styles/PropertyCard.css";

const PropertyCard = (props) => {
  const { title, type, bedrooms, bathrooms, price, city, email } = props;
  return (
    <div className="property-card">
      <div className="property-card__item">
        <b>{title}</b>
      </div>
      <div className="property-card__item">Property Type: {type}</div>
      <div className="property-card__item">
        <FontAwesomeIcon className="icon" icon={faBed} /> {bedrooms}
      </div>
      <div className="property-card__item">
        <FontAwesomeIcon className="icon" icon={faShower} /> {bathrooms}
      </div>
      <div className="property-card__item">Â£{price}</div>
      <div className="property-card__item">City: {city}</div>
      <div className="property-card__email">
        <FontAwesomeIcon icon={faEnvelope} />
        <a href={`mailto:${email}`}> Email</a>
      </div>
    </div>
  );
};

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  bathrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

export default PropertyCard;
