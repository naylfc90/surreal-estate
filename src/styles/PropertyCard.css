.icon {
  padding-right: 5px;
}

.property-card {
  background-color: lightgray;
  padding: 10px;
  border: 1px solid black;
  border-radius: 10px;
  width: 250px;
}

.property-card__item {
  padding-bottom: 5px;
}

.property-card__email {
  padding: 5px;
  background-color: lightskyblue;
  border: 1px solid black;
  border-radius: 10px;
  width: fit-content;
}
