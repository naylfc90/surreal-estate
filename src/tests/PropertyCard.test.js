import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "2 Bedroom Flat",
    type: "Flat",
    bedrooms: 2,
    bathrooms: 1,
    price: 100000,
    city: "Liverpool",
    email: "joe@bloggs.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bedrooms={validProps.bedrooms}
        bathrooms={validProps.bathrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
