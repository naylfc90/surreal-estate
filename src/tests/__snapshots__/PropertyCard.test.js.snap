// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-card__item"
    >
      <b>
        2 Bedroom Flat
      </b>
    </div>
    <div
      class="property-card__item"
    >
      Property Type: Flat
    </div>
    <div
      class="property-card__item"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed icon"
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 288C220.1 288 256 252.1 256 208S220.1 128 176 128S96 163.9 96 208S131.9 288 176 288zM544 128H304C295.2 128 288 135.2 288 144V320H64V48C64 39.16 56.84 32 48 32h-32C7.163 32 0 39.16 0 48v416C0 472.8 7.163 480 16 480h32C56.84 480 64 472.8 64 464V416h512v48c0 8.837 7.163 16 16 16h32c8.837 0 16-7.163 16-16V224C640 170.1 597 128 544 128z"
          fill="currentColor"
        />
      </svg>
       2
    </div>
    <div
      class="property-card__item"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-shower icon"
        data-icon="shower"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 384c-17.67 0-32 14.33-32 32c0 17.67 14.33 32 32 32s32-14.33 32-32C320 398.3 305.7 384 288 384zM416 256c-17.67 0-32 14.33-32 32c0 17.67 14.33 32 32 32s32-14.33 32-32C448 270.3 433.7 256 416 256zM480 192c-17.67 0-32 14.33-32 32c0 17.67 14.33 32 32 32s32-14.33 32-32C512 206.3 497.7 192 480 192zM288 320c0-17.67-14.33-32-32-32s-32 14.33-32 32c0 17.67 14.33 32 32 32S288 337.7 288 320zM320 224c-17.67 0-32 14.33-32 32c0 17.67 14.33 32 32 32s32-14.33 32-32C352 238.3 337.7 224 320 224zM384 224c17.67 0 32-14.33 32-32c0-17.67-14.33-32-32-32s-32 14.33-32 32C352 209.7 366.3 224 384 224zM352 320c-17.67 0-32 14.33-32 32c0 17.67 14.33 32 32 32s32-14.33 32-32C384 334.3 369.7 320 352 320zM347.3 91.31l-11.31-11.31c-6.248-6.248-16.38-6.248-22.63 0l-6.631 6.631c-35.15-26.29-81.81-29.16-119.6-8.779L170.5 61.25C132.2 22.95 63.65 18.33 21.98 71.16C7.027 90.11 0 114.3 0 138.4V464C0 472.8 7.164 480 16 480h32C56.84 480 64 472.8 64 464V131.9c0-19.78 16.09-35.87 35.88-35.87c9.438 0 18.69 3.828 25.38 10.5l16.61 16.61C121.5 160.9 124.3 207.6 150.6 242.7L144 249.4c-6.248 6.248-6.248 16.38 0 22.63l11.31 11.31c6.248 6.25 16.38 6.25 22.63 0l169.4-169.4C353.6 107.7 353.6 97.56 347.3 91.31z"
          fill="currentColor"
        />
      </svg>
       1
    </div>
    <div
      class="property-card__item"
    >
      Â£100000
    </div>
    <div
      class="property-card__item"
    >
      City: Liverpool
    </div>
    <div
      class="property-card__email"
    >
      <a
        href="mailto:joe@bloggs.com"
      >
        Email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
